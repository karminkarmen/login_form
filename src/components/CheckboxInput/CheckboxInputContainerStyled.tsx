import styled from 'styled-components';
import { centerized } from '../../configs/mixins';

const CheckboxInputContainerStyled = styled.div`
    ${centerized}
    margin: ${props => props.theme.margins.normal};
`;

export default CheckboxInputContainerStyled;