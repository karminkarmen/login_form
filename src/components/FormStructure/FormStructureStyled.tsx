import styled from 'styled-components';
import { centered } from '../../configs/mixins';

const FormStructureStyled = styled.div`
    ${centered}
    flex-direction: column;
    border-radius:  ${props => props.theme.radius.maximum};
    width: ${props => props.theme.width.mainElementWidth};
    height: ${props => props.theme.width.mainElementWidth};
`;

export default FormStructureStyled;