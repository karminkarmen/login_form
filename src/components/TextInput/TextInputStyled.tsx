import styled from 'styled-components';

type Props = {
    label: string, 
    type: string
};

const InputStyled = styled.input.attrs((props: Props) => ({
    type: props.type,
    name: props.label,
    id: props.label
}))<Props>`
    font-size: ${props => props.theme.fontSizes.medium};
    border-radius: ${props => props.theme.radius.medium};
    border: 2px solid ${props => props.theme.colors.main};
    padding: ${props => props.theme.paddings.medium};
    background: transparent;
    color: ${props => props.theme.colors.secondaryLightest};
    margin-bottom: 1rem;

    ::placeholder {
        color: ${props => props.theme.colors.background};
    }

    &:focus {
        outline: none;
        background: ${props => props.theme.colors.secondaryLighten};
        color: ${props => props.theme.colors.main};
    }
`;

export default InputStyled;