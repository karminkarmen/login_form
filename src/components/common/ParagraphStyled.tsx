import styled from 'styled-components';

const ParagraphStyled = styled.p`
    margin: 0 0 ${props => props.theme.margins.normal} 0;
    max-width: ${props => props.theme.width.innerElementWidth};
    color: ${props => props.theme.colors.main};
    font-size: ${props => props.theme.fontSizes.normal};
    font-family: ${props => props.theme.fontFamilies.lato};
`;

export default ParagraphStyled;