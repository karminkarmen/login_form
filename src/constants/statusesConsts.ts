export enum loginStatuses {
    AUTHORIZED = 'AUTHORIZED',
    NOT_LOGGED_IN ='NOT_LOGGED_IN',
    REJECTED = 'REJECTED',
}