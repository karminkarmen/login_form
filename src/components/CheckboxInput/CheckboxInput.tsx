import React from 'react';
import { ThemeProvider } from 'styled-components';
import theme from '../../configs/theme';
import LabelStyled from '../common/LabelStyled';
import CheckboxInputContainerStyled from './CheckboxInputContainerStyled';

export type Props = {
    name: string;
    label: string;
}

const CheckboxInput: React.FC<Props> = ({ name, label }) => (
    <ThemeProvider theme={theme}>
        <CheckboxInputContainerStyled>
            <input type="checkbox" name={name} id={name} />
            <LabelStyled 
                target={name} 
                isUppercased={false} 
            >
                {label}
            </LabelStyled>
        </CheckboxInputContainerStyled> 
    </ThemeProvider>
);

export default CheckboxInput;
