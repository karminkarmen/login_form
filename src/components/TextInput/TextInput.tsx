import React from 'react';
import { ThemeProvider } from 'styled-components';
import theme from '../../configs/theme';
import TextInputStyled from './TextInputStyled';
import LabelStyled from '../common/LabelStyled';
import ParagraphStyled from '../common/ParagraphStyled';

type Props = {
    handler: (e: React.ChangeEvent<HTMLInputElement>) => void;
    label: string;
    type: string;
    validationInfo: string;
}

const TextInput: React.FC<Props> = ({ handler, label, type, validationInfo }) => {
    return (
        <ThemeProvider theme={theme}>
            <LabelStyled 
                target={label}
                isUppercased={true}
            >
                {label}
            </LabelStyled>
            <TextInputStyled 
                onChange={handler} 
                type={type} 
                label={label}
            />
            {validationInfo.length > 0 && <ParagraphStyled>{validationInfo}</ParagraphStyled>} 
        </ThemeProvider>
    );
};

export default TextInput;