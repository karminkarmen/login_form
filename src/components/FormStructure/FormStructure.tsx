import React from 'react';
import { ThemeProvider } from 'styled-components';
import theme from '../../configs/theme';
import FormContainerStyled from './FormStructureStyled';
import FieldsetStyled from '../common/FieldsetStyled';
import LoginButton from '../LoginButton/LoginButton';
import TextInput from '../TextInput/TextInput';
import { loginStatuses } from '../../constants/statusesConsts';
import ParagraphStyled from '../common/ParagraphStyled';
import CheckboxInput from '../CheckboxInput/CheckboxInput';

type Props = {
    formSubmitHandler: (e: any) => void, 
    onEmailChange: (e: React.ChangeEvent<HTMLInputElement>) => void,
    onPasswordChange: (e: React.ChangeEvent<HTMLInputElement>) => void,
    emailValidationInfo: string
    passwordValidationInfo: string,
    loginStatus: loginStatuses,
}

const FormStructure: React.FC<Props> = ({ 
    formSubmitHandler, 
    onEmailChange, 
    emailValidationInfo, 
    onPasswordChange, 
    passwordValidationInfo, 
    loginStatus 
}) => {
    return (
        <ThemeProvider theme={theme}>
            <form 
                method="POST" 
                action="" 
                onSubmit={formSubmitHandler} 
            >
                <FieldsetStyled>
                    <FormContainerStyled theme={theme}>
                        <TextInput 
                            handler={onEmailChange} 
                            label={'email'} 
                            type={'text'} 
                            validationInfo={emailValidationInfo} 
                        />
                        <TextInput 
                            handler={onPasswordChange} 
                            label={'password'} 
                            type={'password'} 
                            validationInfo={passwordValidationInfo} 
                        />
                        <CheckboxInput 
                            name={'remember'}
                            label={'Remember me'}
                        />
                        { loginStatus === loginStatuses.REJECTED && <ParagraphStyled>Invalid email or password</ParagraphStyled>}
                    </FormContainerStyled>
                    <LoginButton />
                </FieldsetStyled>
            </form> 
        </ThemeProvider>
    );
};

export default FormStructure;


