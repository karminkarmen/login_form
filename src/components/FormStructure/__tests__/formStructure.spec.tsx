import React from 'react';
import { shallow } from 'enzyme';

import FormStructure from '../FormStructure';
import { LoginStatus } from '../../../constants/statusesConsts';
import TextInput from '../../TextInput/TextInput';
import CheckboxInput from '../../CheckboxInput/CheckboxInput';
import ParagraphStyled from '../../common/ParagraphStyled';
import messages from '../../../constants/messagesConsts';

describe('<FormStructure />', () => {

    const props: React.ComponentProps<typeof FormStructure> = {
        formSubmitHandler: (e) => {}, 
        onEmailChange: (e) => {}, 
        onPasswordChange: (e) => {}, 
        emailValidationInfo: 'Mail validation info',
        passwordValidationInfo: 'Password validation info',
        loginStatus: LoginStatus.NOT_LOGGED_IN,
    };

    it('matches snapshot', () => {
        const wrapper = shallow(<FormStructure {...props} />);
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it('should render form tag with post method', () => {
        const wrapper = shallow(<FormStructure {...props} />);

        expect(wrapper.find(`form[method='POST']`)).toHaveLength(1);
    });

    it('should render 2 text inputs and one checkbox input', () => {
        const wrapper = shallow(<FormStructure {...props} />);

        expect(wrapper.find(TextInput)).toHaveLength(2);
        expect(wrapper.find(CheckboxInput)).toHaveLength(1);
    });

    it('should not render additional not authorized message', () => {
        const wrapper = shallow(<FormStructure {...props} />);

        expect(wrapper.find(ParagraphStyled)).toHaveLength(0);
    });

    it('should render additional message if login status is rejected', () => {
        props.loginStatus = LoginStatus.REJECTED;

        const wrapper = shallow(<FormStructure {...props} />);

        expect(wrapper.find(ParagraphStyled)).toHaveLength(1);

        const paragraph = wrapper.find(ParagraphStyled);
        expect(paragraph.text()).toEqual(messages.notAuthorizedInfo);
    });

});
