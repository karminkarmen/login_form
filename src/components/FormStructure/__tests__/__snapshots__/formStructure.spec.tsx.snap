// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormStructure /> matches snapshot 1`] = `
"<ThemeProvider theme={{...}}>
  <form method=\\"POST\\" action=\\"\\" onSubmit={[Function: formSubmitHandler]}>
    <styled.fieldset>
      <styled.div theme={{...}}>
        <TextInput handler={[Function: onEmailChange]} label=\\"email\\" type=\\"text\\" validationInfo=\\"Mail validation info\\" />
        <TextInput handler={[Function: onPasswordChange]} label=\\"password\\" type=\\"password\\" validationInfo=\\"Password validation info\\" />
        <CheckboxInput name=\\"remember\\" label=\\"Remember me\\" />
      </styled.div>
      <LoginButton />
    </styled.fieldset>
  </form>
</ThemeProvider>"
`;
